### PR TITLE
Add Protobuf and OpenSSL 1.1.0g to Linux x86

### PR DESCRIPTION
**Pull request by vsebe**
Required by OpenJ9 for JITasS (out-of-process JIT compilation)

Issue: #493

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by sxa555** [comment]
@vsebe Just to check - is there a reason why you only included RHEL/CentOS6 and Ubu 16.04 in these playbooks, so it won't get picked up on RHEL7/SLES/Later Ubuntu?

Other than that I'm happy with integrating this (Got someone who might test this for me tomorrow :-) )

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+###################
+# OpenSSL v1.1.0g #
+###################
+# Required by OpenJ9 for out-of-process JIT compilation aka JITasS
```
Comment (sxa555) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: Minor point but should this have been `JITaaS` rather than `JITasS`?

Comment (vsebe) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: right

**Comment by vsebe** [comment]
I haven't yet test it on all platforms (waiting for testing). I'll extend it to all xlinux platforms.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml (hunk 1 above)
Comment (vsebe) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: fixed

**Comment by vsebe** [comment]
@sxa555 - extended to centos, rhel, ubuntu and sles

**Comment by vsebe** [comment]
@sxa555 - ready for your review, thank you!
I have tested the playbooks on ub 14 & 16, centos 6, rhel 6 & 7, sles 12.

**Comment by sxa555** [comment]
I'd rather not have openssl 110 installed directly into `/usr/local` since this may not be a permanent requirement and I want to be easily able to delete it later, so can we use `./config --prefix=/usr/local/openssl110` instead please? I'd also prefer not to have the symlink into `/usr/local` since I want absolutely zero risk of anything picking this up by default.

I'm ok with protobuf being in the default location though :-)

**Comment by sxa555** [comment]
FYI openssl 110g compiles ok on Linux for `ppc64le` and `s390x` too

**Comment by vsebe** [comment]
@sxa555 - if opnessl is installed in a custom location (e.g /usr/local/openssl110), this location would have to be added to the PATH for the OpenJ9 for JITaaS builds. As there are multiple openssl installations this would have to be 1st on the PATH. Removing the symlinks would require to add /usr/local/openssl110/lib64 or /usr/local/openssl110/lib to LD_LIBRARY_PATH.

**Comment by sxa555** [comment]
Yep that sounds reasonable to me

**Comment by vsebe** [comment]
@sxa555 - added --prefix configure option and removed symlinks - ready for your review, thank you!

**Comment by sxa555** [comment]
Just one more things based on some extra understanding obtained today:

(1) Since I've just realised this won't be in the mainstream codebase for some time can you add a comment at the start of the two new files saying something like `Currently only available in the alternate openj9 branch at https://github.com/eclipse/openj9/tree/jitaas` to make it clear why it's in there but not currently used by the mainstream build. 

Tempted to suggest renaming openssl to openssl110g just to be clear, but I'm ok to approve without that since at the moment we're not installing another version (111 was a possibility, but we're bundling that instead now)

**Comment by vsebe** [comment]
@sxa555 - files amended (added requested comment) - ready for your review, thank you!

**Comment by sxa555** [comment]
@vsebe Travis check has failed due to trailing whitespace on the line I asked you to add - can you fix please?

**Pull request merged**